pkg/server/endpoints/node/handler.go

```diff
@@ -878,6 +885,29 @@ func (h *Handler) getBundle(ctx context.Context, trustDomainID string) (*common.
 	return resp.Bundle, nil
 }
 
+type CSR struct {
+	SpiffeID  string
+	PublicKey crypto.PublicKey
+}
+
+func (h *Handler) parseAttestCSR(csrBytes []byte) (*CSR, error) {
+	if h.c.AllowAgentlessNodeAttestors {
+		return parseCSR(csrBytes, idutil.AllowAnyInTrustDomain(h.c.TrustDomain.Host))
+	}
+	return parseCSR(csrBytes, idutil.AllowTrustDomainAgent(h.c.TrustDomain.Host))
+}
+
+func (h *Handler) parseX509CACSR(csrBytes []byte) (*CSR, error) {
+	csr, err := parseCSR(csrBytes, idutil.AllowTrustDomain(h.c.TrustDomain.Host))
+	if err != nil {
+		return nil, err
+	}
+	if csr.SpiffeID == "" {
+		return nil, errors.New("X509 CA CSR is missing the SPIFFE ID")
```

Comment:
Is this not caught with the idutil assertion above?

Reply:
no, `parseCSR` also handles CSR's without the URI.